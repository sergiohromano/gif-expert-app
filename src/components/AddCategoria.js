import React, { useState } from "react";
import { PropTypes } from "prop-types";

export const AddCategoria = ({ setCategories }) => {
    const [inputValue, setInputValue] = useState("");

    const handleSubmit = (e) => {
        if (inputValue.trim().length > 2) {
            e.preventDefault();
            setCategories((cats) => [inputValue, ...cats]);
            setInputValue("");
        }
    };

    return (
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                value={inputValue}
                onChange={(e) => setInputValue(e.target.value)}
            />
        </form>
    );
}

AddCategoria.propTypes = {
    setCategories: PropTypes.func.isRequired,
}