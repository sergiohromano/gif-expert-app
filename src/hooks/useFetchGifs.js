import { useState, useEffect } from "react";
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs = (category) => {
    const [data, setData] = useState([]);
    const [loading, setLoading] = useState(true);

    useEffect(() => {
        getGifs(category)
            .then(setData)
            .then(() => setLoading(false));
    }, [category]);

    return {
        data, 
        loading
    };
}
