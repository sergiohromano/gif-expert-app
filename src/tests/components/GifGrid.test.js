import React from 'react'
import { shallow } from 'enzyme'
import { GifGrid } from './../../components/GifGrid'
import { useFetchGifs } from './../../hooks/useFetchGifs'
jest.mock('./../../hooks/useFetchGifs')


describe('Pruebas en <GifGridItem />',() => {
    const category = 'One Punch';
    
    test('debe de mostrar el component correctamente',() => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: false
        })
        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de mostrar items cuando se cargan iimÃ¡genes useFetchGifs',() => {
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/imagen.jpg',
            title: 'Cualquier cosa'
        },{
            id: 'DEF',
            url: 'https://localhost/imagen.jpg',
            title: 'Cualquier cosa'
        }];
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })
        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper.find('p').exists()).toBe(false)
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
    })
})