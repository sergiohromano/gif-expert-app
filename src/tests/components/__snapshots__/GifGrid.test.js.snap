// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe de mostrar el component correctamente 1`] = `
<div>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  />
</div>
`;
